build/helper/metadata_add_all.py

```diff
@@ -395,10 +401,173 @@ def add_all_enum_metadata(enums, config):
+        f = config['functions'][f_name]
+        for p in f['parameters']:
+            start_enum = None
+            if 'documentation' not in p:
+                continue
+            if 'enum' in p:
+                start_enum = p['enum']
+            _check_documentation(p['documentation'], config, start_enum)
+
+        if 'documentation' not in f:
+            continue
+        start_enum = None
+        if 'enum' in f:
+            start_enum = f['enum']
+        _check_documentation(f['documentation'], config, start_enum)
+
+    # Check attribute documentation
+    for a_id in config['attributes']:
+        a = config['attributes'][a_id]
+        if 'documentation' not in a:
+            continue
+        start_enum = None
+        if 'enum' in a:
+            start_enum = a['enum']
+        _check_documentation(a['documentation'], config, start_enum)
+
+    # Check enum documentation
+    for e_name in config['enums']:
+        e = config['enums'][e_name]
+        if 'documentation' not in e:
+            continue
+        _check_documentation(e['documentation'], config)
+        for v in e['values']:
+            if 'documentation' not in v:
+                continue
+            _check_documentation(v['documentation'], config)
+
+
+def _check_tables(nd):
+    if 'table_header' not in nd and 'table_body' not in nd:
+        return  # We don't need to do anything
+
+    # First we get max length
+    max_len = 0
+    table_header = nd['table_header'] if 'table_header' in nd else None
+    table_body = nd['table_body']
+    if table_header:
+        max_len = len(table_header)
+    for line in table_body:
+        if len(line) > max_len:
+            max_len = len(line)
+
+    # Now make sure all lines have the same number of cells
+    if table_header:
+        while len(table_header) != max_len:
+            table_header.append('')
+    for line in table_body:
+        while len(line) != max_len:
+            line.append('')
+
+
+def _square_up_tables(config):
+    # First we go through the function and parameter documentation
+    for f_name in config['functions']:
+        f = config['functions'][f_name]
+        for p in f['parameters']:
+            if 'documentation' not in p:
+                continue
+            _check_tables(p['documentation'])
+
+        if 'documentation' not in f:
+            continue
+        _check_tables(f['documentation'])
+
+    # Check attribute documentation
+    for a_id in config['attributes']:
+        a = config['attributes'][a_id]
+        if 'documentation' not in a:
+            continue
+        _check_tables(a['documentation'])
+
+    # Check enum documentation
+    for e_name in config['enums']:
+        e = config['enums'][e_name]
+        if 'documentation' not in e:
+            continue
+        _check_tables(e['documentation'])
+        for v in e['values']:
+            if 'documentation' not in v:
+                continue
+            _check_tables(v['documentation'])
```

Comment:
Consider moving documentation specific functions to documentation_helper.py.

Reply:
`add_notes_re_links` and `square_up_tables` plus helper functions moved to documentation_helper.py